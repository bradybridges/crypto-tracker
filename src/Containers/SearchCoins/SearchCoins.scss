input {
  width: 300px;
  height: 75px;
  border-radius: 4px;
  border: solid black 1px;
  text-align: center;
  font-size: 1.2em;
}

#search-header {
  text-align: center;
}

#search-coins-container {
  height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../../Images/30\ Black\ abstract\ Wallpapers\ HD\ Download.jpg');
  color: white;
  justify-content: space-evenly;
}

#search-button-div {
  background-color: black;
  width: 100px;
  margin: 0 auto;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  width: 300px;
  height: 70px;
  border: solid white 1px;
}

#search-form {
  width: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 30vh;
}
