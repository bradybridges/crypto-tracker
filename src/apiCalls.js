import { apiKey } from './apiKey';

export const fetchTopCryptos = async () => {
  const url = 'https://api.nomics.com/v1/currencies/ticker';
  const key = `?key=${apiKey}`;
  const queries = '&ids=BTC,ETH,XRP,LTC,XMR,ETC,BCH&interval=1h,1d,30d,365d&convert=USD'
  const response = await fetch(`${url}${key}${queries}`);
  if(!response.ok) {
    throw Error('Failed to fetch coins...');
  } else {
    const coins = await response.json();
    return coins;
  }
}
