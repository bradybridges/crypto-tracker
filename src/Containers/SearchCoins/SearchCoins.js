import React, { Component } from 'react';
import './SearchCoins.scss';
import searchImg from '../../Images/search.svg';
import { searchCoin } from '../../apiCalls';
import CoinCard  from '../CoinCard/CoinCard';

class SearchCoins extends Component {
  constructor() {
    super();
    this.state = {
      symbol: '',
      coin: {},
    }
  }

  handleChange = (e) => {
    const symbol = e.target.value.toUpperCase();
    this.setState({[e.target.name]: symbol});
  }

  handleCoinSearch = async (e) => {
    e.preventDefault();
    const symbol = this.state.symbol;
    const coin = await searchCoin(symbol);
    if(coin.length === 0) {
      console.log(coin[0]);
      this.setState({symbol: '', coin: {error: 'Coin Not Found'}});
    } else {
      this.setState({coin: coin[0]});
      this.setState({symbol: ''});
    }
  }

  render() {
    const {rank, logo_url, price, name, circulating_supply} = this.state.coin;
    return(
      <section id='search-coins-container'>
        <form id='search-form' onSubmit={this.handleCoinSearch}>
          <h2 id='search-header'>Search For Coins By Symbol</h2>
          <input name='symbol' type='text' maxLength='4' placeholder='Ex BTC' value={this.state.symbol} onChange={this.handleChange} />
          <button id='search-button-div'>
            <img src={searchImg} alt='search button' />
          </button>
        </form>
        {this.state.coin.name && <CoinCard  getCryptos={this.props.getCryptos} coin={this.state.coin} name={name} rank={rank} price={price} logo={logo_url} circulatingSupply={circulating_supply} />}
        {this.state.coin.error && <h3>Coin Not Found</h3>}
      </section>
    );
  }
}

export default SearchCoins;