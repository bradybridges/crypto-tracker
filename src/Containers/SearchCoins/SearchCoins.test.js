import React from 'react';
import { shallow } from 'enzyme';
import SearchCoins from './SearchCoins';
import * as api from '../../apiCalls';

describe('SearchCoins', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchCoins />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should update state with symbol', () => {
    const mockEvent = {
      target: { value: 'btc', name: 'symbol' },
    };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('symbol')).toEqual('BTC');
  });

  it.skip('handleCoinSearch should update state of coin if coin is found and reset symbol', () => {
    api.searchCoin = jest.fn().mockImplementation(() => {
      return [{
        currecy: 'BTC',
        id: 'BTC',
        price: 9341.6987,
        price_date: '2019-11-02',
        symbol: 'BTC',
        circulating_supply: '18025852',
        max_supply: '21000000',
        name: 'Bitcoin',
        logo_url: 'https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg',
        market_cap: '1243423423.05',
        rank: '1',
        high: '19480.90',
        high_timestamp: '2017-12-16',
      }];
    });
    const mockEvent = {
      preventDefault: jest.fn(),
    };
    wrapper.setState({symbol: 'BTC'});
    wrapper.instance().handleCoinSearch(mockEvent);
  });
});