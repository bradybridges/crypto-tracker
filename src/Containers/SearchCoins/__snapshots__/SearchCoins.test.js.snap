// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCoins should match the snapshot 1`] = `
<section
  id="search-coins-container"
>
  <form
    id="search-form"
    onSubmit={[Function]}
  >
    <h2
      id="search-header"
    >
      Search For Coins By Symbol
    </h2>
    <input
      maxLength="4"
      name="symbol"
      onChange={[Function]}
      placeholder="Ex BTC"
      type="text"
      value=""
    />
    <div
      id="search-button-div"
    >
      <img
        alt="search button"
        src="search.svg"
      />
    </div>
  </form>
</section>
`;
